packages/site/cms-site/src/sitePreview/SitePreviewApiHelper.ts

```diff
@@ -0,0 +1,71 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+type Scope = Record<string, unknown>;
+type GraphQLClient = {
+    setHeader(key: string, value: string): unknown;
+    request(document: string, variables?: unknown): Promise<{ isAllowedSitePreview: boolean }>;
+};
```

Comment:
@nsams this basically adds a dependency on graphql-request to `@comet/cms-site` (yes, you could use another client, but who will do that?), so we can ignore this in #1681/#1682?

Have you thought about alternative simply POST-ing to the API?

Reply:
That typing is Ok (the alternative no typing at all isn't better)

Naming it GraphQLClient but only returning isAllowedSitePreview is strange though. (but it is not exported, so no issue)